tools/ports/sdl3.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2014 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import glob
+import os
+
+from tools import diagnostics
+
+TAG = 'release-3.2.4'
+HASH = 'c26a8afeec481e3ae3b435eec405d9f99d78752ebf5118963cd56728ceff23772769f5291df581329488da7489034e835301b08d61a42c811764e24b3542a4c2'
+SUBDIR = 'SDL-' + TAG
+
+variants = {'sdl3-mt': {'PTHREADS': 1}}
+
+
+def needed(settings):
+  return settings.USE_SDL == 3
+
+
+def get_lib_name(settings):
+  return 'libSDL3' + ('-mt' if settings.PTHREADS else '') + '.a'
+
+
+def get(ports, settings, shared):
+  # get the port
+  ports.fetch_project('sdl3', f'https://github.com/libsdl-org/SDL/archive/{TAG}.zip', sha512hash=HASH)
+
+  def create(final):
+    diagnostics.warning('experimental', 'sdl3 port is still experimental')
+    root_dir = ports.get_dir('sdl3', SUBDIR)
+
+    # copy includes to a location so they can be used as 'SDL3/'
+    source_include_path = os.path.join(root_dir, 'include', 'SDL3')
+    ports.install_headers(source_include_path, target='SDL3')
+
+    # In addtion copy our pregenerated SDL_build_config.h file.
+    here = os.path.dirname(__file__)
+    ports.install_headers(os.path.join(here, 'sdl3'), 'SDL_build_config.h', target='SDL3')
+
+    glob_patterns = [
+      # Generic sources (from SDL3's CMakeLists.txt)
+      '*.c',
+      'atomic/*.c',
+      'audio/*.c',
+      'camera/*.c',
+      'core/*.c',
+      'cpuinfo/*.c',
+      'dynapi/*.c',
+      'events/*.c',
+      'io/*.c',
+      'io/generic/*.c',
+      'filesystem/*.c',
+      'gpu/*.c',
+      'joystick/*.c',
+      'haptic/*.c',
+      'hidapi/*.c',
+      'locale/*.c',
+      'main/*.c',
+      'misc/*.c',
+      'power/*.c',
+      'render/*.c',
+      'render/*/*.c',
+      'sensor/*.c',
+      'stdlib/*.c',
+      'storage/*.c',
+      'thread/*.c',
+      'time/*.c',
+      'timer/*.c',
+      'video/*.c',
+      'video/yuv2rgb/*.c',
+      'tray/*.c',
+      # Platform speecifc sources
+      'storage/generic/*.c',
+      'tray/unix/*.c',
+      'time/unix/*.c',
+      'timer/unix/*.c',
+      'main/emscripten/*.c',
+      'filesystem/posix/*.c',
+      'filesystem/emscripten/*.c',
+      'locale/emscripten/*.c',
+      'camera/emscripten/*.c',
+      'joystick/emscripten/*.c',
+      'joystick/virtual/*.c',
+      'power/emscripten/*.c',
+      'misc/emscripten/*.c',
+      'audio/emscripten/*.c',
+      'video/emscripten/*.c',
+      'video/offscreen/*.c',
+      'audio/disk/*.c',
+      'loadso/dlopen/*.c',
+      # Dummy backends, do we need to support these?
```

Comment:
@slouken, sorry for another dumb question, but do you think its worth including the dummy backends?    I guess it will increase code size, but maybe allow SDL3 to be used in more cases/places?

Reply:
I'm not sure how useful they are for emscripten, maybe if someone needs to disable audio or something like that? @icculus, thoughts?